controllers/tempo/microservices_controller.go

```diff
@@ -36,6 +41,16 @@ type MicroservicesReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+type DegradedError struct {
+	Reason  v1alpha1.ConditionReason
+	Message string
+	Requeue bool
+}
+
+func (e *DegradedError) Error() string {
+	return fmt.Sprintf("cluster degraded: %s", e.Message)
```

Comment:
How about adding the `Reason` to the message?

Reply:
Updated.